unic/ucd/normal/src/canonical_combining_class.rs

```diff
@@ -12,44 +12,98 @@
 
 //! Accessor for Canonical_Combining_Class (ccc) property
 
-// TODO: [API] Support Canonical_Combining_Class_Values, either by creating a separate `enum` type,
-// or converting `CanonicalCombiningClass` into enum and allowing access to numerical value.
-// <http://unicode.org/reports/tr44/#Canonical_Combining_Class_Values>
-
-// TODO: Use associate type `u8`
 
 use std::cmp::Ordering;
 
 
 /// Represents *Canonical_Combining_Class* property of a Unicode character.
 ///
 /// * <http://unicode.org/reports/tr44/#Canonical_Combining_Class>
-pub trait CanonicalCombiningClass {
-    /// If the *ccc* has value `Not_Reordered` (`0`).
-    fn ccc_is_not_reordered(&self) -> bool;
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct CanonicalCombiningClass(u8);
+
+
+// TODO: Once we fully adopt 1.20 change these to associated consts on CanonicalCombiningClass
+/// Canonical Combining Classes by their name
+#[allow(non_upper_case_globals)]
+pub mod values {
+    use super::CanonicalCombiningClass as CCC;
+
+    /// Spacing and enclosing marks; also many vowel and consonant signs, even if nonspacing
+    pub const NotReordered: CCC = CCC(0);
+    /// Marks which overlay a base letter or symbol
+    pub const Overlay: CCC = CCC(1);
+    /// Diacritic nukta marks in Brahmi-derived scripts
+    pub const Nukta: CCC = CCC(7);
+    /// Hiragana/Katakana voicing marks
+    pub const KanaVoicing: CCC = CCC(8);
+    /// Viramas
+    pub const Virama: CCC = CCC(9);
+    /// Marks attached at the bottom left
+    pub const AttatchedBelowLeft: CCC = CCC(200);
+    /// Marks attached directly below
+    pub const AttachedBelow: CCC = CCC(202);
+    /// Marks attached at the bottom right
+    pub const AttachedBelowRight: CCC = CCC(204);
```

Comment:
Sound reasonable, specially since it's not *the long-name*, but a local naming *based on* the long name.

Okay, let's go ahead as it is and we expand the general API guideline to fit this in.